### PR TITLE
Improved script to check for shopify inventory and continue selling check before making checkInventory call and enabling bopis(#2khjfe8)

### PR DESCRIPTION
**Pull request by ymaheshwari1**
Dependent on #24 
- Added condition to check for shopify inventory and continue selling before enabling bopis app
- Updated condition to add comma in store address.

### CODE REVIEW

File: scripts/script-tag.js [hunk 1]
```diff
@@ -95,48 +95,51 @@
...
-            })
-        })
+    async function getCurrentProduct() {
+        await jQueryBopis.getJSON(`${window.location.pathname}.js`, function(product) {
+            currentProduct = product;
+        });
+    }
+
+    async function isProductAvailable(variantSku) {
+        const hasInventoryOnShopify = jQueryBopis("input[class='hc_inventory']").val() > 0
```
Comment (adityasharma7) [inline scripts/script-tag.js]:
```suggestion
        const hasInventoryOnShopify = jQueryBopis("input.hc_inventory").val() > 0
```

Comment (adityasharma7) [reply scripts/script-tag.js]: Check if it is not breaking when element is not found

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: scripts/script-tag.js [hunk 2]
```diff
@@ -238,11 +263,7 @@
         let storeInformation = await getStoreInformation(queryString).then(data => data).catch(err => err);
         let result = '';
 
-        const id = jQueryBopis(".hc-product-form").serializeArray().find(ele => ele.name === "hc-product-sku").value
-        
-        // when using the demo instance we will use id as sku, and for dev instance we will use sku
-        // const sku = meta.product.variants.find(variant => variant.id == id).sku
-        const sku = id;
+        const sku = jQueryBopis("input[class='hc_product_sku']").val();
```
Comment (adityasharma7) [inline scripts/script-tag.js]: Use class selector

File: scripts/script-tag.js [hunk 3]
```diff
@@ -94,31 +95,51 @@
...
             await getCurrentLocation();
 
             jQueryBopis(".hc-store-information").remove();
-            jQueryBopis(".hc-open-bopis-modal").remove();
             jQueryBopis(".hc-bopis-modal").remove();
 
             // TODO Simplify this [name='id']. There is no need to serialize
             const cartForm = jQueryBopis(".hc-product-form");
-            const id = cartForm.serializeArray().find(ele => ele.name === "hc-product-sku").value;
+            const sku = jQueryBopis("input[class='hc_product_sku']").val();
```
Comment (adityasharma7) [inline scripts/script-tag.js]: Use class selector

File: scripts/script-tag.js [hunk 4]
```diff
@@ -94,31 +95,51 @@
...
+
+            // Do not enable BOPIS when the current product is not available
+            if(!(await isProductAvailable(sku))) return;
 
             const bopisButton = jQueryBopis("#hc-bopis-button");
+            const bopisButtonEnabled = jQueryBopis("#hc-bopis-button > button");
```
Comment (adityasharma7) [inline scripts/script-tag.js]:
```suggestion
            const existingBopisButton = jQueryBopis("#hc-bopis-button > button");
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: scripts/script-tag.js (hunk 1 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Sir, when the element is not found then the statement will return false and will not break.

File: scripts/script-tag.js (hunk 2 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Done Sir

File: scripts/script-tag.js (hunk 3 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Done Sir

File: scripts/script-tag.js (hunk 4 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Done Sir

File: scripts/script-tag.js [hunk 5]
```diff
@@ -94,31 +95,51 @@
...
+    async function isProductAvailable(variantSku) {
+        const hasInventoryOnShopify = jQueryBopis("input.hc_inventory").val() > 0
+        if (currentProduct) {
+            if (hasInventoryOnShopify) {
+                return true;
+            } else {
+                return currentProduct.variants.find((variant) => variant.sku == variantSku).inventory_policy === 'continue'
+            }
+        }
+        return false;
```
Comment (adityasharma7) [inline scripts/script-tag.js]:
```suggestion
        const hasInventoryOnShopify = jQueryBopis("input.hc_inventory").val() > 0
        return currentProduct && (hasInventoryOnShopify || currentProduct.variants.find((variant) => variant.sku == variantSku).inventory_policy === 'continue')
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: scripts/script-tag.js (hunk 5 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Sir, Improved the condition.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**